synapse/axon.py

```diff
@@ -1009,6 +1041,37 @@ async def readlines(self, sha256):
         if remain:
             yield remain
 
+    @staticmethod
+    def _getCsvReader(lines, dialect, **fmtparams) -> csv.reader:
+        try:
+            reader = csv.reader(lines, dialect, **fmtparams)
+        except TypeError as e:
+            if 'invalid keyword' in str(e):
+                raise s_exc.BadArg(mesg=f'Invalid csv format parameter: {str(e)}') from None
+            else:  # pragma: no cover
+                raise
+        else:
+            return reader
+
+    async def csvrows(self, sha256, dialect='excel', **fmtparams):
+        dialect = dialect.lower()
+        if dialect not in csv.list_dialects():
+            raise s_exc.BadArg(mesg=f'Invalid CSV dialect, use one of {csv.list_dialects()}')
+
+        lines = []
+        async for line in self.readlines(sha256):
+            lines.append(line)
+            if len(lines) >= 64:
```

Comment:
This is a little bit of a magic number here.
At some point we'll have captured a sufficient number of lines and want to start yielding them through the reader object.

Reply:
We should create a sync version of readlines and hand it into the csv reader as a generator

Reply:
We can feed line chunks one at a time to a queue that a csvreader can handle iterating over, so we don't need a sync version of readlines.